docs/source/DeveloperGuide/DevelopmentEnvironment/DeveloperInstallation/editingCode.rst

```diff
@@ -4,222 +4,250 @@
 =================
 
 Here we describe the suggested method for editing and unit-testing DIRAC code, and direct extensions of it.
 
-
 What is this NOT for?
 =====================
 
 * This method is NOT specific for the WebAppDIRAC or Pilot code development, although several things that are described here can be applied.
-* This method will NOT work out of box if you need 3rd party python libraries that are not pip installable.
-
-
+* There will be no explanation of how to use Git. If you're unfamiliar with Git there are many excellent resources online. A particularly nice interactive tutorial can be found at `learngitbranching.js.org <https://learngitbranching.js.org/>`_.
 
 Notes before continuing
 =======================
 
-*OS*: any \*nix should be fine
-(maybe even windows is fine but I would not know how).
-Examples below are given for Linux (and specifically, the writer used Ubuntu)
-
-*shell*: examples below are given in bash, and are proven to work also in zsh.
-Any csh like tcsh should not pose a problem.
-
-*repository*: as already explained,
-DIRAC's GIT repositories are hosted on `GitHub <https://github.com/DIRACGrid>`_.
-for which you need to have an account before continuing.
+*OS*: Linux or macOS should both be fine.
+If you wish to develop on Windows the easiest solution is probably to use the `Windows Subsystem for Linux <https://docs.microsoft.com/en-us/windows/wsl/>`_.
 
+*shell*: These instructions will work out of the box in any bourne compatible shell (bash/zsh/ksh) but other shells, such as fish/csh/tcsh, will work with only minor adjustments.
 
+*repository*: DIRAC's GIT repositories are hosted on `GitHub <https://github.com/DIRACGrid>`_ and you will need a GitHub account to contribute.
 
 Stuff you need to have installed
 ================================
 
-*python*: make sure python 2.7.9+ (possibly 2.7.13) is installed and set as default
-(beware: the latest versions of Ubuntu use python 3.X as default, SLC6 use python 2.6 as default).
+*git*: DIRAC's version control system of choice is git and most computers will already have this installed. If not, you can easily install it by searching online for your operating system name and "install git".
 
-*python-pip*: the tool for installing python packages hosted
-on `PyPi <https://pypi.python.org/pypi>`_.
+*editor*: Almost any text editor is fine so if you already have a preference for vim/emacs/sublime/eclipse/pycharm/atom use that.
+If you're new to development, Visual Studio Code is free modern editor with an extensive plugin ecosystem and many nice IDE like features.
+Regardless of which editor you choose it's worth investigating what plugins are available plugins.
 
-*git*: DIRAC's version control system of choice is git, so install it.
+*python*: DIRAC is written in Python so you will need to have Python environment available. While there are many possible ways of doing this the simplest option for most people is to use either conda or DIRACOS2 (Linux only).
```

Comment:
```suggestion
*python*: DIRAC is written in Python so you will need to have Python environment available. While there are many possible ways of doing this, the simplest option for most people is to use either `conda <https://docs.conda.io/en/latest/>`_ or `DIRACOS2 <https://github.com/DIRACGrid/DIRACOS2>`_ (Linux only).
```

Reply:
I'm inclined not to add the links here as there likely to cause confusion (e.g. install the Anaconda distribution and then get a mess of conflicts or subtle ABI incompatibilities). How about:
```suggestion
*python*: DIRAC is written in Python so you will need to have a Python environment available. While there are many possible ways of doing this, the simplest option for most people is to use either conda or DIRACOS2 (Linux only). Details about how to use both options are included below.
```

Reply:
Fine for me.